flytepropeller/pkg/controller/handler.go

```diff
@@ -103,7 +110,8 @@ func (p *Propeller) TryMutateWorkflow(ctx context.Context, originalW *v1alpha1.F
 	ctx = contextutils.WithResourceVersion(ctx, mutableW.GetResourceVersion())
 
 	maxRetries := uint32(p.cfg.MaxWorkflowRetries)
-	if IsDeleted(mutableW) || (mutableW.Status.FailedAttempts > maxRetries) {
+
+	if IsAborted(mutableW) || (mutableW.Status.FailedAttempts > maxRetries) {
```

Comment:
we could still have the case that the CRD gets deleted by some other process and should still account for that here.

Reply:
mmmmh could be? Looking inside the if, the function there looks like is only for aborting case

Reply:
I added back the isDeleted(...) to manage this case here too, now it is backward compatible, wdyt?